docs/modules/ROOT/pages/explanation/decisions/night-maintenance.adoc

```diff
@@ -0,0 +1,35 @@
+= Offering Nighttime Maintenance
+
+== Problem
+
+Maintenance of APPUiO Cloud is currently done during daytime hours.
+Although we encourage and support our customers to deploy their applications in a highly available configuration,
+this has led to significant dissatisfaction among some customers.
+
+Many have expressed a preference for potentially disruptive node reboots to occur after office hours.
+This presents a challenge as we have applications on APPUiO Cloud that are dependencies for other clusters maintenance,
+necessitating the maintenance schedule to remain primarily as is to avoid impacting these interdependent systems.
+
+== Proposals
+
+Two potential solutions have been proposed to address customer dissatisfaction while also considering the operational requirements of APPUiO Cloud:
+
+
+=== Implementation of a New Node-Class for Nighttime Reboots
+This solution involves creating a new node-class for APPUiO Cloud that would be designated for reboots during the night.
+The machine config pool for these nodes would be paused during the day and unpaused at night to facilitate this maintenance schedule.
```

Comment:
```suggestion
To implement this, we'd enhance the https://github.com/appuio/openshift-upgrade-controller/[upgrade controller] to allow specifying custom maintenance windows for individual machine config pools.
The upgrade controller would then make sure that any machine config pools which have a custom maintenance window are paused as the maintenance starts.
Additionally, the upgrade controller would ensure that the machine config pools are unpaused as their maintenance window begins.
Most likely, we'd implement the custom window specification as an offset from the maintenance start.
```

Reply:
Should we include implementation details in the decission?